core/src/test/java/io/grpc/internal/RetriableStreamTest.java

```diff
@@ -683,6 +688,48 @@ public void messageAvailable() {
     verify(masterListener).messagesAvailable(messageProducer);
   }
 
+  @Test
+  public void closedWhileDraining() {
+    ClientStream mockStream1 = mock(ClientStream.class);
+    doReturn(mockStream1).when(retriableStreamRecorder).newSubstream();
+
+    final ClientStream mockStream3 = mock(ClientStream.class);
+    final ClientStream mockStream2 =
+        mock(
+            ClientStream.class,
+            delegatesTo(
+                new NoopClientStream() {
+                  @Override
+                  public void start(ClientStreamListener listener) {
+                    // closed while draning and retry
+                    doReturn(true).when(retriableStreamRecorder).shouldRetry();
```

Comment:
Changing stub return values within this start seems weird, as I'm not sure why they are related. It also makes the test harder to follow. Why isn't this just done early in `closedWhileDraining()`? It doesn't look like it ever will return `false` now.

Reply:
I wouldn't do this if mockito doReturn supported varargs. And I don't like `when(a.method()).thenReturn(b).thenReturn(c)`